core/src/test/resources/sql/measure.iq

```diff
@@ -179,6 +179,53 @@ from empm;
 
 !ok
 
+# [CALCITE-6559] LogicalCorrelate Type mismatch
+# Occurs when AVG measure is applied to SMALLINT or TINYINT column.
+# A CAST is added internally to the result of SUM / COUNT, but this cast
+# prevents a cast to nullable for the null-generating side of an outer join.
+WITH empm AS (
+    SELECT *, AVG(empno) AS MEASURE avg_empno
+    FROM emp)
+SELECT job, avg_empno
+FROM empm
+GROUP BY job;
++-----------+-----------+
+| JOB       | AVG_EMPNO |
++-----------+-----------+
+| ANALYST   |      7845 |
+| CLERK     |      7769 |
+| MANAGER   |      7682 |
+| PRESIDENT |      7839 |
+| SALESMAN  |      7629 |
++-----------+-----------+
+(5 rows)
+
+!ok
+
+!if (false) {
```

Comment:
Is there an issue filed for this other problem?

Reply:
I've logged https://issues.apache.org/jira/browse/CALCITE-6564